### ISSUE TITLE
`unnecessary-default-type-args (UP043)` isn't triggering on stubs

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#20011]
### Summary

`UP043` doesn't trigger at all on typeshed. I guess because it doesn't ignore th version in stubs? (relates to https://github.com/astral-sh/ruff/issues/19566)
```
(typeshed) PS ...\typeshed> uv run ruff check --select=UP043
All checks passed!
```

### Version

ruff 0.12.9

**Comment by ntBre** [comment]
Yep, looks like it always requires 3.13+:

https://github.com/astral-sh/ruff/blob/2245355931f66def488809a6b9b784f7c294bd49/crates/ruff_linter/src/checkers/ast/analyze/expression.rs#L144-L148

I think the fix would be as easy as adding `|| checker.semantic().in_stub_file()` here, if these are always valid in stub files.
